### ISSUE TITLE
SSR is not working. Fedora 32, RX 580 with amdgpu driver, 1920 x 1080 monitor

### ISSUE THREAD

**Issue by philthegitter** [MaartenBaert/ssr#835]
Hi, I've been using it for years, thanks very much for this wonderful app. The machine is a dual Xeon E5-2696v3 tower computer, each cpu has 18 physical cores, so it has 36 physical cores plus hyperthreading for a total of 72 cores seen by the app. The machine has 32 GB of ram. Video card is an MSI Armor RX 580 with 8 GB of memory. Monitor is an Asus VG248QE connected through a DisplayPort cable, DISPLAY=:0.0, running at its native 1920 x 1080 resolution. I've tried ssr both from the official Fedora 32 repo and by compiling it from git source using simple-build-and-install. I've tried it when running the monitor at both 60Hz and 144Hz (switched using xrandr). When I try to run start capture, I get

[PageRecord::StartPage] Starting page ...
[PageRecord::StartPage] Started page.
[PageRecord::StartOutput] Starting output ...
[Muxer::Init] Using format matroska (Matroska).
[Muxer::AddStream] Using codec libx264 (libx264 H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10).
[VideoEncoder::PrepareStream] Using pixel format nv12.
[libx264 @ 0x9d06c0] Application has requested 72 threads. Using a thread count greater than 16 is not recommended.
[libx264 @ 0x9d06c0] using SAR=1/1
[libx264 @ 0x9d06c0] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
[libx264 @ 0x9d06c0] profile High, level 3.0, 4:2:0, 8-bit
[libx264 @ 0x9d06c0] 264 - core 159 r2999 296494a - H.264/MPEG-4 AVC codec - Copyleft 2003-2020 - http://www.videolan.org/x264.html - options: cabac=1 ref=1 deblock=1:0:0 analyse=0x3:0x3 me=dia subme=1 psy=1 psy_rd=1.00:0.00 mixed_ref=0 me_range=16 chroma_me=1 trellis=0 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=0 threads=72 lookahead_threads=2 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=1 keyint=250 keyint_min=25 scenecut=40 intra_refresh=0 rc=crf mbtree=0 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 pb_ratio=1.30 aq=1:1.00
[Muxer::AddStream] Using codec libvorbis (libvorbis).
[BaseEncoder::EncoderThread] Encoder thread started.
[AudioEncoder::PrepareStream] Using sample format f32p.
[BaseEncoder::EncoderThread] Encoder thread started.
[Muxer::MuxerThread] Muxer thread started.
[PageRecord::StartOutput] Started output.
[Synchronizer::SynchronizerThread] Synchronizer thread started.
[PageRecord::StartInput] Starting input ...
[X11Input::Init] Using X11 shared memory.
[X11Input::InputThread] Input thread started.
[PulseAudioInput::InputThread] Stream is a monitor.
[PageRecord::StartInput] Started input.
[PulseAudioInput::InputThread] Input thread started.
[X11Input::InputThread] Error: Can't get image (using shared memory)!
    Usually this means the recording area is not completely inside the screen. Or did you change the screen resolution?
[X11Input::InputThread] Exception 'X11Exception' in input thread.
[FastResampler::Resample] Resample ratio is 1.0000 (was 0.0000).

Help appreciated. Thanks in advance.
Best regards,

Philip Courier

**Comment by MaartenBaert** [comment]
Are you running a standard X11 session? SSR currently doesn't support Wayland and doesn't work properly with some non-standard X11 servers (e.g. VNC) or if you try to run it through SSH.

**Comment by philthegitter** [comment]
Thanks for your reply, sir. Yes, it's a standard X11 session, it's the official xfce spin of Fedora 32. No VNC, no SSH, no Wayland. I've tried lowering the number of active cores to 4 (physical) per cpu with no luck, and I've also tried reverting to an R 290 video card using just the radeon driver (modprobe.blacklist=amdgpu in the grub config, so amdgpu does not show up in lsmod), with the same unfortunate result. I have tried installing ssr into a live-cd (the installation .iso dd'ed onto a usb stick) session of Fedora 32 and in the live-cd session ssr works perfectly. So the failure must be due to something I've installed since the initial Fedora 32 installation. I'm a longtime Linux user but not cognoscenti enough to know where to look from here for clues. Suggestions appreciated. Thanks much.

**Comment by MaartenBaert** [comment]
Can you please upload your SSR settings file (`~/.ssr/settings.conf`)?

Have you checked that the recorded area x/y/width/height parameters are correct and match the actual resolution of the screen? I.e. they don't fall outside the screen area?

**Comment by philthegitter** [comment]
It's too late--I already reinstalled Fedora 32. Simplescreenrecorder is working perfectly now.

I suspect the trouble arose when I played with connecting two monitors in side-by-side Xinerama fashion. After the commands

xrandr --setmonitor OverlayDisplay 3840/1062x1080/299+0+0 DVI-D-0,DisplayPort-0
xrandr --output DVI-D-0 --mode 1920x1080 --rate 60.00 --output DisplayPort-0 --mode 1920x1080 --rate 144.00 --right-of DVI-D-0

I did have the two monitors presenting a side-by-side combined single desktop. I reverted it with

xrandr --delmonitor OverlayDisplay
xrandr --output DisplayPort-0 --mode 1920x1080 --rate 144.00 --primary --screen 0
xrandr --output DVI-D-0 --off
xrandr --output DVI-D-0 --mode 1920x1080 --rate 60.00 --same-as DisplayPort-0

and afterward noticed something odd. With just the DisplayPort monitor plugged into the video card, all was fine, except, simplescreenrecorder would not work, with the error described above. 

When I tried reconnecting the DVI-D-0 monitor, it was still displaying the left-hand half of the 3840 x 1080 combined wide desktop.

All efforts using xrandr to force the DVI-D-0 monitor to display the same desktop as the DisplayPort monitor...worked only during the current session. Upon reboot, the DVI-D-0 monitor stubbornly reverted to the left-half-of-wide-desktop display. This, despite putting appropriate xrandr commands in /etc/lightdm/lightdm.conf and then I tried in /etc/lightdm/lightdm.conf.d/10-monitor.conf. Always, no matter what I tried, the xrandr "duplicate the DisplayPort monitor on the DVI-D-0 monitor" command took effect only until the very late stages of the bootup process. But then, after the desktop was fully displayed and just before desktop presentation was fully complete, the DVI-D-0 monitor would revert to the left-half-of-wide-desktop presentation.

I tried using kfind to see if there was any file within /etc which contained an xrandr command countermanding the ones I put in /etc/lightdm .conf files, and kfind could not find any.

My newbie wild guess is, somehow either pam or the session manager was reverting the X display settings to the wide display, in the late stages of bootup.

Whatever the case, a fresh reinstall of Fedora 32 (official xfce4 spin) has restored simplescreenrecorder to smooth functioning.

Best regards, and thanks again for this wonderful utility!
